landingai/pipeline/frameset.py

```diff
@@ -41,18 +38,36 @@ class PredictionList(List[ClassificationPrediction]):
     False
     """
 
+    def __init__(self, *args: Any) -> None:
+        # TODO: This is a hack to support OCR predictions. We should probably have different PredictionList
+        #  classes for each type of prediction, so we don't have to do conditional checks everywhere and `cast`
+        # or "type: ignore".
+        super().__init__(*args)
+        for p in self:
+            if not isinstance(p, type(self[0])):
+                raise ValueError("All elements should be of the same type")
+
+    @property
+    def _inner_type(self) -> str:
+        return type(self[0]).__name__
+
     def __contains__(self, key: object) -> bool:
         if not len(self):
             return False
         if isinstance(key, str):
-            # TODO: This is a hack to support OCR predictions. We should probably have different PredictionList
-            #  classes for each type of prediction, so we don't have to do this kind of checks and "type: ignore"
-            if all(isinstance(p, OcrPrediction) for p in self):
+            if self._inner_type == "OcrPrediction":
                 # For OCR predictions, check if the key is in the full text
-                full_text = " ".join([p.text for p in self])  # type: ignore
+                full_text = ""
+                for p in self:
+                    p = cast(OcrPrediction, p)
+                    full_text += " " + p.text
                 return key in full_text
             else:
-                return any([p.label_name == key for p in self])
+                for p in self:
+                    p = cast(ClassificationPrediction, p)
+                    if p.label_name == key:
+                        return True
+                return False
```

Comment:
Ah, sorry. I forgot to comment about this changes: iterating in C (list comprehention, str.join with generator, etc) is usually way faster than iterating in Python's `for`. Please, don't change this implementation:

```
In [1]: def find_with_for(x):
   ...:     for i in values:
   ...:         if i == x:
   ...:             return True
   ...:     return False
   ...: 

In [2]: def find_with_any(x):
   ...:     return any((x == 500 for i in values))
   ...: 

In [3]: import timeit

In [4]: values = list(range(1000))

In [5]: timeit.timeit("find_with_for(500)", globals=globals())
Out[5]: 6.868485922997934

In [6]: timeit.timeit("find_with_any(500)", globals=globals())

Out[6]: 0.4467220990009082
```


```
In [1]: def join_string():
   ...:     return " ".join((str(i) for i in values))
   ...: 

In [2]: def join_with_for():
   ...:     ret = ""
   ...:     for i in values:
   ...:         ret += " " + str(i)
   ...:     return ret
   ...: 

In [3]: values = list(range(100))

In [4]: import timeit

In [5]: timeit.timeit("join_string()", globals=globals())
Out[5]: 8.381673152995063

In [6]: timeit.timeit("join_with_for()", globals=globals())
Out[6]: 11.856949225002609
```

Reply:
Right, I changed it because if the casting, but you helped me realize I could use `cast` without loosing the generator approach.